shotover-proxy/src/frame/cassandra.rs

```diff
@@ -43,6 +46,22 @@ pub(crate) fn metadata(bytes: &[u8]) -> Result<CassandraMetadata> {
     })
 }
 
+/// Count queries only from an unparsed Cassandra frame
+pub(crate) fn get_message_count(bytes: &[u8]) -> Result<NonZeroU32> {
+    let frame = RawCassandraFrame::from_buffer(bytes, Compression::None)
```

Comment:
Now that we have to actually allocate for the message body its not completely clear that these sorts of speed hacks are still worth it.
If we allocate here and then later on call frame() we will be allocating for the body twice.

Possibly we should have a way to get the body without allocation.
Possibly we should rewrite CQL to only parse when needed.
Happy to land this as is at the moment but eventually we will need to measure this kind of thing and figure out how far we should go with these kinds of speed hacks.

Reply:
I'm happy to do it in this PR.

Reply:
I think writing some benchmarks that just hit the RequestThrottling transform (in a prereq PR) would be a good place to start.
That would help evaluate the performance impact of each approach.

Reply:
Opened a PR for benchmarks: #565

Reply:
Resolving as benchmarks PR has been merged.